# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RDDProcessHost.h
# Commit: dd05584049c6
# Full Hash: dd05584049c6e6138dd7dfe6067ebd03e1219782
# Author: Michael Froman <mfroman@mozilla.com>
# Date: 2019-04-10 09:58:56
# Regressor Bug: 1539029
# File Overlap Count: 1
# Description:
#   Bug 1539029 - pt 4 - add shared pref serializer/deserializer to RDD process. r=kmag
#   
#   Depends on D26568
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D26569
# ==============================================================================

diff -r fcab7823cf27 -r dd05584049c6 dom/media/ipc/RDDProcessHost.h
--- a/dom/media/ipc/RDDProcessHost.h	Tue Apr 09 21:07:47 2019 +0000
+++ b/dom/media/ipc/RDDProcessHost.h	Tue Apr 09 21:08:00 2019 +0000
@@ -13,6 +13,11 @@
 #include "mozilla/ipc/ProtocolUtils.h"
 #include "mozilla/ipc/TaskFactory.h"
 
+namespace mozilla {
+namespace ipc {
+class SharedPreferenceSerializer;
+}
+}
 class nsITimer;
 
 namespace mozilla {
@@ -139,6 +144,8 @@
   UniquePtr<RDDChild> mRDDChild;
   uint64_t mProcessToken;
 
+  UniquePtr<ipc::SharedPreferenceSerializer> mPrefSerializer;
+
   bool mShutdownRequested;
   bool mChannelClosed;
 